[gengyudong] iP

gengyudong opened the pull request:
# GengPro
> "GengPro is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

GengPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Manage tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gengyudong committed:
Complete Level 1: Echo

gengyudong committed:
Complete Level-2: Add, List

gengyudong committed:
Complete Level-3: Mark as Done

gengyudong committed:
Complete Level-4: ToDo, Event, Deadline

gengyudong committed:
Complete A-TextUiTesting: Automated Text UI Testing

gengyudong committed:
Complete Level-5: Handle Errors

gengyudong committed:
Complete Level-6: Delete

gengyudong committed:
Add feature to save tasks onto hard disc

gengyudong committed:
Add feature of understanding datetime

gengyudong committed:
Amend the code to align with OOP

gengyudong committed:
Put files into packages

gengyudong committed:
Merge branch 'add-gradle-support'

gengyudong committed:
Add JUnit tests

gengyudong committed:
Create a JAR File

gengyudong committed:
Add Javadoc comments for all files

gengyudong committed:
Merge branch 'branch-A-JavaDoc'

gengyudong committed:
Adhere to coding standard

gengyudong committed:
Merge branch 'branch-A-CodingStandard'

gengyudong committed:
Minor code standard change

gengyudong committed:
Add feature: find keyword in task list

gengyudong committed:
Merge branch 'branch-Level-9'

subiloble commented on src/main/java/duke/ui/Storage.java:
```diff
@@ -47,6 +63,15 @@ public ArrayList<Task> load() throws GengException {
         return taskList;
     }
 
+    /**
+     * Parses a task from a line in the file. The line format is expected to be:
+     * - T | isDone | description
+     * - D | isDone | description | deadline
+     * - E | isDone | description | start-end
+     *
+     * @param line The line to be parsed into a Task object.
+     * @return A Task object if parsing is successful, otherwise null.
+     */
```
I like your JavaDoc comments for the method which is clear and well-defined.

subiloble commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+package duke.commands;
+
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.GengException;
+import duke.ui.Storage;
+import duke.ui.Ui;
```
Perhaps it would be better to rename the root folder to geng or other related names since the app is no longer duke.

subiloble reviewed (commented):
LGTM.

dnardnar commented on src/main/java/duke/Geng.java:
```diff
@@ -0,0 +1,70 @@
+package duke;
+
+import duke.commands.Command;
+import duke.tasks.TaskList;
+import duke.ui.*;
+
```
Consider renaming the package duke to something more relevant, such as geng, to align with the application's current name. This would improve clarity and consistency with the project. Perhaps import your ui classes one by one to follow the coding standard
```suggestion
package dnar;

import dnar.commands.Command;
import dnar.tasks.TaskList;
import dnar.ui.''
import dnar.ui.''
import dnar.ui.''
```

dnardnar commented on src/main/java/duke/tasks/Events.java:
```diff
@@ -0,0 +1,79 @@
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm a");
+        return "E |" + super.toString() + " | " + this.startDatetime.format(formatter) + " - " +
+                this.endDatetime.format(formatter);
```
I like how you broke these long lines into multiple lines, adhering to the 120-character limit. This improves readability and aligns well with the coding standard for line length. Great job!

dnardnar commented on src/main/java/duke/ui/Storage.java:
```diff
@@ -0,0 +1,137 @@
+ * updated tasks back to the file.
+ */
+public class Storage {
+    private final String FILE_PATH;
+
+    /**
+     * Constructor for Storage that initializes the file path.
+     *
+     * @param filePath The path to the file where tasks will be saved or loaded from.
+     */
+    public Storage(String filePath) {
+        this.FILE_PATH = filePath;
+    }
```
I like how you named this constant in all uppercase with underscores separating words, following the Screaming Snake Case convention. It aligns perfectly with the coding standard for constants. Great job!

dnardnar commented on build.gradle:
```diff
@@ -0,0 +1,41 @@
+    archiveClassifier = null
+}
+
+run{
+    standardInput = System.in
+}
```
```suggestion
run {
    standardInput = System.in
}
```
I like how you followed the coding standard by using proper indentation and spacing for the run block. However, I noticed that this specific line is missing one space between run and {. Adding the space would make it consistent with the rest of the codebase, which adheres well to the standard. Great job overall!

dnardnar reviewed (commented):
LGTM. Just a few nits to fix.

gengyudong committed:
Style fix: Fix style issues for all files

gengyudong committed:
Add Javafx

gengyudong committed:
Add GUI for geng chatbot

gengyudong committed:
Fix code to adhere to coding guidelines

gengyudong committed:
Generate new .jar file

gengyudong committed:
Add assert statements to enforce critical assumptions

Currently, the code lacks explicit checks for key assumptions, making it
harder to catch unexpected issues during runtime.

Add assert statements to validate user input and method parameters.

Using Java's assert feature helps document assumptions clearly while
keeping the code efficient, as assertions can be enabled or disabled
as needed.

Assertions improve maintainability and debugging without affecting
production performance when disabled.

gengyudong committed:
Code standard update: updated variable names and javadoc

The previous code had some errors in variable naming and javadoc
formatting.

Changed the name of some variables and standardize the structure of
javadoc comments.

The changes ensure that the code is aligned with the coding standard.

No functionality changes, this commit focuses solely on coding
conventions.

gengyudong committed:
Merge pull request #2 from gengyudong/branch-A-Assertions

Add assert statements to enforce critical assumptions

gengyudong committed:
Merge branch 'master' into branch-A-CodeQuality

gengyudong committed:
Merge pull request #3 from gengyudong/branch-A-CodeQuality

Branch a code quality

gengyudong committed:
Add gradle.yml

gengyudong committed:
Add feature: Show schedule

gengyudong committed:
Add more JUnit tests

gengyudong committed:
Merge branch 'branch-A-MoreTesting'

# Conflicts:
#	src/test/java/geng/ui/ParserTest.java
#	src/test/java/tasks/TaskListTest.java

gengyudong committed:
bug resolve: make exceptions appear on GUI

gengyudong committed:
Improve the GUI

gengyudong committed:
Edit user guide and fix some bugs

gengyudong committed:
Add background colour to dialog box
